### PR TITLE
refactor - Bulk management tab unit tests

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Cleans up BulkManagement component and adds docstrings and unit tests.

BulkManagement becomes BulkManagementTab (because that's what it is)
4 new sub-components: BulkManagementAlerts, FileUploadForm, HistoryTable, and ResultsSummary.

the `transfofrmHistoryEntry=>getRowsProcessed` pipeline is *only* used for this table, so I cleaned up its output to explicitly give what is needed for the results summary, instead of just passing the values necessary to derive said string, when the values themselves aren't actually needed.

Partial ticket completion
JIRA: [AU-3](https://openedx.atlassian.net/browse/AU-3)

**Developer Checklist**
- [ ] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**
The main place to check that is still working is the BulkManagement tab, and then just a smoke test of basic functionality.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by nsprenkle — APPROVED** [review]
Code and functionality looks good, but this does introduce some style changes that I want to make sure are intentional:
1. New "Upload Grade CSV" label not previously present
2. Different icons in "Download Summary"
3. Hints/page messages no longer have punctuation

File: src/components/BulkManagementTab/__snapshots__/BulkManagementAlerts.test.jsx.snap [hunk 1]
```diff
@@ -12,7 +12,7 @@ exports[`BulkManagementAlerts component no errer, no upload success snapshot - b
     show={false}
     variant="success"
   >
-    CSV processing. File uploads may take several minutes to complets
+    CSV processing. File uploads may take several minutes to complets.
```
Comment (nsprenkle) [inline src/components/BulkManagementTab/__snapshots__/BulkManagementAlerts.test.jsx.snap:15]: Not introduced by you, but we should fix the typo anyways:

```suggestion
    CSV processing. File uploads may take several minutes to complete.
```

File: src/components/BulkManagementTab/__snapshots__/BulkManagementAlerts.test.jsx.snap [hunk 2]
```diff
@@ -31,7 +31,7 @@ exports[`BulkManagementAlerts component no errer, no upload success snapshot - d
     show={true}
     variant="success"
   >
-    CSV processing. File uploads may take several minutes to complets
+    CSV processing. File uploads may take several minutes to complets.
```
Comment (nsprenkle) [inline src/components/BulkManagementTab/__snapshots__/BulkManagementAlerts.test.jsx.snap:34]: Same typo here:
```suggestion
    CSV processing. File uploads may take several minutes to complete.
```

File: src/components/BulkManagementTab/__snapshots__/test.jsx.snap [hunk 3]
```diff
@@ -3,7 +3,7 @@
 exports[`BulkManagementTab component snapshot snapshot - loads heading from messages.BulkManagementTab.heading, <BulkManagementAlerts />, <FileUploadForm />, <HistoryTable /> 1`] = `
 <div>
   <h4>
-    Use this feature by downloading a CSV for bulk management, overriding grades locally and coming back here to upload
+    Use this feature by downloading a CSV for bulk management, overriding grades locally and coming back here to upload.
```
Comment (nsprenkle) [inline src/components/BulkManagementTab/__snapshots__/test.jsx.snap:6]: Nit: should we include that Oxford comma?

```suggestion
    Use this feature by downloading a CSV for bulk management, overriding grades locally, and coming back here to upload.
```

File: src/data/constants/app.js [hunk 4]
```diff
@@ -57,12 +57,12 @@ export const bulkManagementColumns = [
 export const messages = StrictDict({
   BulkManagementTab: StrictDict({
     csvUploadLabel: 'Upload Grade CSV',
-    heading: 'Use this feature by downloading a CSV for bulk management, overriding grades locally and coming back here to upload',
+    heading: 'Use this feature by downloading a CSV for bulk management, overriding grades locally and coming back here to upload.',
```
Comment (nsprenkle) [inline src/data/constants/app.js:60]: Same note here about oxford comma

```suggestion
    heading: 'Use this feature by downloading a CSV for bulk management, overriding grades locally, and coming back here to upload.',
```

File: src/data/constants/app.js [hunk 5]
```diff
@@ -57,12 +57,12 @@ export const bulkManagementColumns = [
...
     importBtnText: 'Import Grades',
-    successDialog: 'CSV processing. File uploads may take several minutes to complets',
+    successDialog: 'CSV processing. File uploads may take several minutes to complets.',
```
Comment (nsprenkle) [inline src/data/constants/app.js:62]: Same typo here
```suggestion
    successDialog: 'CSV processing. File uploads may take several minutes to complete.',
```

**Review by nsprenkle — COMMENTED** [review]
A note on some typos that have been here probably forever but we might as well fix

File: src/containers/GradebookPage/index.jsx [hunk 6]
```diff
@@ -63,12 +64,16 @@ export class GradebookPage extends React.Component {
...
 };
 GradebookPage.propTypes = {
+  history: PropTypes.shape({
+    push: PropTypes.func,
+  }).isRequired,
   location: PropTypes.shape({ search: PropTypes.string }),
   courseId: PropTypes.string.isRequired,
   initializeApp: PropTypes.func.isRequired,
-  showBulkManagement: PropTypes.bool.isRequired,
+  showBulkManagement: PropTypes.bool,
```
Comment (jansenk) [inline src/containers/GradebookPage/index.jsx:76]: why?

File: src/data/constants/api.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * bulkGradesUrlByCourseAndRow(courseId, rowId)
+ * returns the bulkGrades url with the given courseId and rowId.
+ * @param {string} courseId - course identifier
+ * @param {string} rowId - row/error identifier
+ * @return {string} - bulk grades fetch url
+ */
+export const bulkGradesUrlByCourseAndRow = (courseId, rowId) => (
+  `${baseUrl}/bulkGrades/course/${courseId}/?error_id=${rowId}`
```
Comment (jansenk) [inline src/data/constants/api.js:13]: Not to be pedantic, but this isn't a constant. Should it go somewhere else?

File: src/data/constants/app.js [hunk 8]
```diff
@@ -18,3 +18,51 @@ export const localFilterKeys = StrictDict({
...
+
+/**
+ * column configuration for bulk management tab's data table
+ */
+export const bulkManagementColumns = [
+  {
+    key: 'filename',
+    label: 'Gradebook',
+    columnSortable: false,
+    width: 'col-5',
```
[the remark this answers is not in the document]
Comment (jansenk) [reply src/data/constants/app.js:30]: I agree, but this almost seems like more of an issue with paragon...

File: src/components/BulkManagementTab/ResultsSummary.jsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+ * @param {string} text - summary string
+ */
+const ResultsSummary = ({
+  courseId,
+  rowId,
+  text,
+}) => (
+  <Hyperlink
+    href={bulkGradesUrlByCourseAndRow(courseId, rowId)}
+    destination="www.edx.org"
```
Comment (jansenk) [inline src/components/BulkManagementTab/ResultsSummary.jsx:24]: what?

**Review by jansenk — APPROVED** [review]
Looks god, a couple qs

File: src/containers/GradebookPage/index.jsx (hunk 6 above)
Comment (muselesscreator) [reply src/containers/GradebookPage/index.jsx:76]: can return undefined, which causes error message but not failure.

File: src/data/constants/api.js (hunk 7 above)
Comment (muselesscreator) [reply src/data/constants/api.js:13]: honestly, it should probably go into the LmsApiService file, and may well do so, once I clean that up.  I just really didn't want to go in and muck with that until I was ready to do more thorough overhauls in that file.

File: src/components/BulkManagementTab/ResultsSummary.jsx (hunk 9 above)
Comment (muselesscreator) [reply src/components/BulkManagementTab/ResultsSummary.jsx:24]: in order to user Hyperlink without complaints, we must have a destination, even if not used... is weird but has no functionality impact

**Comment by muselesscreator** [comment]
@jansenk @nsprenkle the col-5 thing is actually a Bootstrap paradigm, fwiw, and has direct JS implications on functionality (how columns get generated), just abstracted away from us a bit.

**Review by mattcarter — APPROVED** [review]
👍  Approved
